### PR TITLE
MINOR: Add javadoc to all ConfigDef.Types values

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java [hunk 1]
```diff
@@ -57,7 +57,7 @@
  * defs.define(&quot;config_with_default&quot;, Type.STRING, &quot;default string value&quot;, Importance.High, &quot;Configuration with default value.&quot;);
  * // check {@link #define(String, Type, Object, Validator, Importance, String)} for more details.
  * defs.define(&quot;config_with_validator&quot;, Type.INT, 42, Range.atLeast(0), Importance.High, &quot;Configuration with user provided validator.&quot;);
- * // check {@link #define(String, Type, Importance, String, String, int, Width, String, List<String>)} for more details.
+ * // check {@link #define(String, Type, Importance, String, String, int, Width, String, List) define(String, Type, Importance, String, String, int, Width, String, List&lt;String&gt;)} for more details.
```
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:60]: This fixes warnings when generating javadoc on this file
```
ConfigDef.java:81: warning - Tag @link:illegal character: "60" in "#define(String, Type, Importance, String, String, int, Width, String, List<String>)"
ConfigDef.java:81: warning - Tag @link:illegal character: "62" in "#define(String, Type, Importance, String, String, int, Width, String, List<String>)"
ConfigDef.java:81: warning - Tag @link: can't find define(String, Type, Importance, String, String, int, Width, String, List<String>) in org.apache.kafka.common.config.ConfigDef
```

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java [hunk 2]
```diff
@@ -802,11 +802,63 @@ else if (value instanceof Class)
...
+        /**
+         * Used for values that implement a Kafka interface. Values must be provided as a Class object or as a
+         * String object, otherwise a {@link ConfigException} is thrown. When the value is provided as a String it must
+         * be the binary name of the Class.
+         */
+        CLASS,
+        /**
+         * Used for string values containing sensitive data such as a password or key. The values of configurations with
+         * of this type are not included in logs and instead replaced with "[hidden]". Values must be provided as a
+         * String object, otherwise a {@link ConfigException} is thrown.
```
Comment (mimaison) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:854]: Technically you can also provide a `org.apache.kafka.common.config.types.Password` object but `Password` is not part of the public API so I omit that option.

File: clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java [hunk 3]
```diff
@@ -802,11 +802,63 @@ else if (value instanceof Class)
...
+        LONG,
+        /**
+         * Used for numerical values within the Java Double range. Values must be provided as a Number object, as a
+         * Double object or as a String being a valid Double value, otherwise a {@link ConfigException} is thrown.
+         */
+        DOUBLE,
+        /**
+         * Used for list values. Values must be provided as a List object, as a String object, otherwise a
+         * {@link ConfigException} is thrown. When the value is provided as a String it must use commas to separate the
+         * different entries (for example: <code>first-entry, second-entry</code>) and an empty String maps to an empty List.
```
Comment (jlprat) [inline clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java:842]: Great to show an example for the String case

**Review by jlprat — APPROVED** [review]
LGTM. Thanks @mimaison

**Pull request merged**